diff -urN aa/k3b-0.11.12/k3b-i18n-0.11/admin/cvs.sh k3b-0.11.12/k3b-i18n-0.11/admin/cvs.sh
--- aa/k3b-0.11.12/k3b-i18n-0.11/admin/cvs.sh	2004-02-29 19:32:00.000000000 +0100
+++ k3b-0.11.12/k3b-i18n-0.11/admin/cvs.sh	2004-07-31 21:18:53.021248896 +0200
@@ -66,7 +66,7 @@
     echo "*** KDE requires automake $required_automake_version"
     exit 1
     ;;
-  automake*1.6.* | automake*1.7* | automake*1.8* ) : ;;
+  automake*1.6.* | automake*1.7* | automake*1.8* | automake*1.9*) : ;;
   "" )
     echo "*** AUTOMAKE NOT FOUND!."
     echo "*** KDE requires automake $required_automake_version"
