--- src/k3b.desktop.orig	2006-08-07 11:03:24.000000000 +0200
+++ src/k3b.desktop	2006-08-07 17:35:19.459909280 +0200
@@ -103,5 +103,5 @@
 MimeType=application/x-k3b
 X-KDE-StartupNotify=true
 X-KDE-NativeMimeType=application/x-k3b;application/x-iso
-Categories=KDE;Application;AudioVideo;DiscBurning;
+Categories=Qt;KDE;Application;AudioVideo;DiscBurning;
 X-DCOP-ServiceType=Unique
--- k3bsetup/k3bsetup2.desktop.orig	2006-08-07 11:03:26.000000000 +0200
+++ k3bsetup/k3bsetup2.desktop	2006-08-07 17:36:04.073127040 +0200
@@ -77,7 +77,7 @@
 X-KDE-Library=k3bsetup2
 X-KDE-ModuleType=Library
 X-KDE-RootOnly=true
-Categories=Application;System;X-KDE-System;
+Categories=Qt;KDE;X-KDE-settings-system;
 GenericName=CD & DVD Burning Setup
 GenericName[ar]=العام  اعداد كتابة القرص المدمج (CD) و القرص المرئي الرقمي (DVD)
 GenericName[bg]=Настройки за запис на CD и DVD
