--- k3b-0.11.18/k3bsetup/k3bsetup2.cpp~	2004-11-28 14:27:24.000000000 +0200
+++ k3b-0.11.18/k3bsetup/k3bsetup2.cpp	2005-01-02 20:03:05.000000000 +0200
@@ -329,7 +329,7 @@
 
   d->config->setGroup( "General Settings" );
   w->m_checkUseBurningGroup->setChecked( d->config->readBoolEntry( "use burning group", false ) );
-  w->m_editBurningGroup->setText( d->config->readEntry( "burning group", "burning" ) );
+  w->m_editBurningGroup->setText( d->config->readEntry( "burning group", "cdwrite" ) );
 
 
   // load search path
@@ -343,7 +343,7 @@
 void K3bSetup2::defaults()
 {
   w->m_checkUseBurningGroup->setChecked(false);
-  w->m_editBurningGroup->setText( "burning" );
+  w->m_editBurningGroup->setText( "cdwrite" );
 
   //
   // This is a hack to work around a kcm bug which makes the faulty assumption that
@@ -486,7 +486,7 @@
 QString K3bSetup2::burningGroup() const
 {
   QString g = w->m_editBurningGroup->text();
-  return g.isEmpty() ? QString("burning") : g;
+  return g.isEmpty() ? QString("cdwrite") : g;
 }
 
 
