--- k3b-2.0.2/plugins/decoder/ffmpeg/k3bffmpegwrapper.cpp~	2011-01-15 21:47:29.000000000 +0100
+++ k3b-2.0.2/plugins/decoder/ffmpeg/k3bffmpegwrapper.cpp	2011-06-30 10:57:03.437376120 +0200
@@ -109,7 +109,7 @@
 #else
     ::AVCodecContext* codecContext =  d->formatContext->streams[0]->codec;
 #endif
-    if( codecContext->codec_type != CODEC_TYPE_AUDIO ) {
+    if( codecContext->codec_type != AVMEDIA_TYPE_AUDIO ) {
         kDebug() << "(K3bFFMpegFile) not a simple audio stream: " << m_filename;
         return false;
     }
