--- k3b-1.0pre1/libk3bdevice/configure.in.in.wig	2006-08-18 14:00:14.000000000 +0200
+++ k3b-1.0pre1/libk3bdevice/configure.in.in	2006-08-18 14:01:33.000000000 +0200
@@ -163,7 +163,7 @@
   dbusqt_lib=NOTFOUND
   dbusqt=NOTFOUND
 
-  search_incs="$kde_includes /usr/include /usr/include/dbus-1.0 /usr/local/include /usr/local/include/dbus-1.0"
+  search_incs="$kde_includes /usr/include /usr/include/dbus-1.0/qt3 /usr/local/include /usr/local/include/dbus-1.0/qt3"
   AC_FIND_FILE(dbus/connection.h, $search_incs, dbusqt_incdir)
 
   if test -r $dbusqt_incdir/dbus/connection.h ; then
@@ -177,10 +177,10 @@
   fi
 
   search_libs="$kde_libraries /usr/lib /usr/lib64 /usr/local/lib"
-  AC_FIND_FILE(libdbus-qt-1.so, $search_libs, dbusqt_libdir)
+  AC_FIND_FILE(libdbus-qt3-1.so, $search_libs, dbusqt_libdir)
 
-  if test -r $dbusqt_libdir/libdbus-qt-1.so ; then
-    DBUSQT_LIBS="-L$dbusqt_libdir -ldbus-qt-1"
+  if test -r $dbusqt_libdir/libdbus-qt3-1.so ; then
+    DBUSQT_LIBS="-L$dbusqt_libdir -ldbus-qt3-1"
     dbusqt_lib=FOUND
   fi
 
