--- k3b-0.12/plugins/decoder/musepack/k3bmpcwrapper.cpp.orig	2005-06-14 04:11:31.000000000 +0200
+++ k3b-0.12/plugins/decoder/musepack/k3bmpcwrapper.cpp	2005-06-14 04:11:56.000000000 +0200
@@ -18,11 +18,6 @@
 #include <kdebug.h>
 #include <qfile.h>
 
-
-#ifndef BOOL
-#define BOOL mpc_bool_t
-#endif
-
 mpc_int32_t read_impl( void* data, void* ptr, mpc_int32_t size )
 {
   QFile* input = static_cast<QFile*>( data );
