--- admin/acinclude.m4.in	2008-05-27 10:23:30.000000000 +0200
+++ admin/acinclude.m4.in-new	2009-07-12 16:17:53.748582894 +0200
@@ -1791,11 +1791,11 @@
 AC_MSG_CHECKING([for KDE])
 
 if test "${prefix}" != NONE; then
-  kde_includes=${includedir}
-  KDE_EXPAND_MAKEVAR(ac_kde_includes, includedir)
+  kde_includes=${includedir}/kde3
+  KDE_EXPAND_MAKEVAR(ac_kde_includes, includedir/kde3)
 
-  kde_libraries=${libdir}
-  KDE_EXPAND_MAKEVAR(ac_kde_libraries, libdir)
+  kde_libraries=${libdir}/kde3dev
+  KDE_EXPAND_MAKEVAR(ac_kde_libraries, libdir/kde3dev)
 
 else
   ac_kde_includes=
@@ -1829,7 +1829,7 @@
 So, check this please and use another prefix!])
 fi
 
-kde_libdirs="$kde_libs_prefix/lib${kdelibsuff} /usr/lib/kde/lib${kdelibsuff} /usr/local/kde/lib${kdelibsuff} /usr/kde/lib${kdelibsuff} /usr/lib${kdelibsuff}/kde /usr/lib${kdelibsuff}/kde3 /usr/lib${kdelibsuff} /usr/X11R6/lib${kdelibsuff} /usr/local/lib${kdelibsuff} /opt/kde3/lib${kdelibsuff} /opt/kde/lib${kdelibsuff} /usr/X11R6/kde/lib${kdelibsuff}"
+kde_libdirs="$kde_libs_prefix/lib${kdelibsuff}/kde3dev /usr/lib/kde/lib${kdelibsuff} /usr/local/kde/lib${kdelibsuff} /usr/kde/lib${kdelibsuff} /usr/lib${kdelibsuff}/kde /usr/lib${kdelibsuff}/kde3 /usr/lib${kdelibsuff} /usr/X11R6/lib${kdelibsuff} /usr/local/lib${kdelibsuff} /opt/kde3/lib${kdelibsuff} /opt/kde/lib${kdelibsuff} /usr/X11R6/kde/lib${kdelibsuff}"
 test -n "$KDEDIR" && kde_libdirs="$KDEDIR/lib${kdelibsuff} $KDEDIR $kde_libdirs"
 kde_libdirs="$ac_kde_libraries $libdir $kde_libdirs"
 AC_FIND_FILE($kde_check_lib, $kde_libdirs, kde_libdir)
@@ -1839,12 +1839,6 @@
 dnl this might be somewhere else
 AC_FIND_FILE("kde3/plugins/designer/kdewidgets.la", $kde_libdirs, kde_widgetdir)
 
-if test -n "$ac_kde_libraries" && test ! -r "$ac_kde_libraries/$kde_check_lib"; then
-AC_MSG_ERROR([
-in the prefix, you've chosen, are no KDE libraries installed. This will fail.
-So, check this please and use another prefix!])
-fi
-
 if test -n "$kde_widgetdir" && test ! -r "$kde_widgetdir/kde3/plugins/designer/kdewidgets.la"; then
 AC_MSG_ERROR([
 I can't find the designer plugins. These are required and should have been installed
@@ -1886,8 +1880,8 @@
   AC_MSG_RESULT([will be installed in $ac_kde_prefix and $ac_kde_exec_prefix])
  fi
 
- kde_libraries="${libdir}"
- kde_includes="${includedir}"
+ kde_libraries="${libdir}/kde3dev"
+ kde_includes="${includedir}/kde3"
 
 else
   ac_cv_have_kde="have_kde=yes \
@@ -1928,7 +1922,7 @@
 KDE_CHECK_UIC_PLUGINS
 fi
 
-ac_kde_libraries="$kde_libdir"
+ac_kde_libraries="$kde_libdir/kde3dev"
 
 AC_SUBST(AUTODIRS)
 
--- admin/ltmain.sh	2009-07-12 16:17:37.648523138 +0200
+++ admin/ltmain.sh-new	2009-07-12 16:17:45.251231198 +0200
@@ -5669,7 +5669,23 @@
 	    case $hardcode_action in
 	    immediate | unsupported)
 	      if test "$hardcode_direct" = no; then
-		add="$dir/$linklib"
+		if test "$linklib" = "libDCOP.so"; then
+			add="$dir/kde3dev/$linklib"
+		elif test "$linklib" = "libkwalletclient.so"; then
+			add="$dir/kde3dev/$linklib"
+		elif test "$linklib" = "libkdefx.so"; then
+			add="$dir/kde3dev/$linklib"
+		elif test "$linklib" = "libkdecore.so"; then
+			add="$dir/kde3dev/$linklib"
+		elif test "$linklib" = "libkdeui.so"; then
+			add="$dir/kde3dev/$linklib"
+		elif test "$linklib" = "libkparts.so"; then
+			add="$dir/kde3dev/$linklib"
+		elif test "$linklib" = "libkio.so"; then
+			add="$dir/kde3dev/$linklib"
+		else
+			add="$dir/$linklib"
+		fi
 		case $host in
 		  *-*-sco3.2v5.0.[024]*) add_dir="-L$dir" ;;
 		  *-*-sysv4*uw2*) add_dir="-L$dir" ;;
--- admin/acinclude.m4.in-new	2009-07-12 16:59:51.390969154 +0200
+++ admin/acinclude.m4.in	2009-07-12 17:00:06.398894540 +0200
@@ -578,7 +578,7 @@
           KDE_FIND_PATH(mcopidl, MCOPIDL, [$kde_default_bindirs], [KDE_MISSING_ARTS_ERROR(mcopidl)])
           KDE_FIND_PATH(artsc-config, ARTSCCONFIG, [$kde_default_bindirs], [KDE_MISSING_ARTS_ERROR(artsc-config)])
         fi
-        KDE_FIND_PATH(meinproc, MEINPROC, [$kde_default_bindirs])
+        KDE_FIND_PATH(meinproc4, MEINPROC, [$kde_default_bindirs])
 
         kde32ornewer=1
         kde33ornewer=1
--- admin/am_edit	2008-05-27 10:23:30.000000000 +0200
+++ admin/am_edit-new	2009-07-12 17:03:38.005110229 +0200
@@ -1999,7 +1999,7 @@
       my $lines = "";
       my $lookup = 'MEINPROC\s*=';
       if ($MakefileData !~ /\n($lookup)/) {
-	$lines = "MEINPROC=/\$(kde_bindir)/meinproc\n";
+	$lines = "MEINPROC=/\$(kde_bindir)/meinproc4\n";
       }
       $lookup = 'KDE_XSL_STYLESHEET\s*=';
       if ($MakefileData !~ /\n($lookup)/) {
--- k3b-i18n-1.0.5/admin/acinclude.m4.in.old	2009-07-12 17:11:48.840906438 +0200
+++ k3b-i18n-1.0.5/admin/acinclude.m4.in	2009-07-12 17:12:01.948410314 +0200
@@ -578,7 +578,7 @@
           KDE_FIND_PATH(mcopidl, MCOPIDL, [$kde_default_bindirs], [KDE_MISSING_ARTS_ERROR(mcopidl)])
           KDE_FIND_PATH(artsc-config, ARTSCCONFIG, [$kde_default_bindirs], [KDE_MISSING_ARTS_ERROR(artsc-config)])
         fi
-        KDE_FIND_PATH(meinproc, MEINPROC, [$kde_default_bindirs])
+        KDE_FIND_PATH(meinproc4, MEINPROC, [$kde_default_bindirs])
 
         kde32ornewer=1
         kde33ornewer=1
--- k3b-i18n-1.0.5/admin/am_edit.old	2009-07-12 17:11:52.847501698 +0200
+++ k3b-i18n-1.0.5/admin/am_edit	2009-07-12 17:12:09.890240661 +0200
@@ -1999,7 +1999,7 @@
       my $lines = "";
       my $lookup = 'MEINPROC\s*=';
       if ($MakefileData !~ /\n($lookup)/) {
-	$lines = "MEINPROC=/\$(kde_bindir)/meinproc\n";
+	$lines = "MEINPROC=/\$(kde_bindir)/meinproc4\n";
       }
       $lookup = 'KDE_XSL_STYLESHEET\s*=';
       if ($MakefileData !~ /\n($lookup)/) {
