diff -uNr k3b-1.0.5/plugins/decoder/ffmpeg.orig/configure.in.in k3b-1.0.5/plugins/decoder/ffmpeg/configure.in.in
--- k3b-1.0.5/plugins/decoder/ffmpeg.orig/configure.in.in	2008-05-27 08:23:36.000000000 +0000
+++ k3b-1.0.5/plugins/decoder/ffmpeg/configure.in.in	2008-10-02 12:12:49.000000000 +0000
@@ -21,8 +21,8 @@
 	AC_LANG_CPLUSPLUS
 	AC_COMPILE_IFELSE( 
 		extern "C" {
-		#include <ffmpeg/avformat.h>
-		#include <ffmpeg/avcodec.h>
+		#include <libavformat/avformat.h>
+		#include <libavcodec/avcodec.h>
 		}
 
 		int main() {
@@ -36,8 +36,8 @@
 	LIBS="-lavformat -lavcodec $LIBS"
 	AC_LINK_IFELSE( 
 		extern "C" {
-		#include <ffmpeg/avformat.h>
-		#include <ffmpeg/avcodec.h>
+		#include <libavformat/avformat.h>
+		#include <libavcodec/avcodec.h>
 		}
 
 		int main() {
diff -uNr k3b-1.0.5/plugins/decoder/ffmpeg.orig/k3bffmpegdecoder.cpp k3b-1.0.5/plugins/decoder/ffmpeg/k3bffmpegdecoder.cpp
--- k3b-1.0.5/plugins/decoder/ffmpeg.orig/k3bffmpegdecoder.cpp	2008-05-27 08:23:36.000000000 +0000
+++ k3b-1.0.5/plugins/decoder/ffmpeg/k3bffmpegdecoder.cpp	2008-10-02 12:13:06.000000000 +0000
@@ -22,7 +22,7 @@
 #include <k3bpluginfactory.h>
 
 extern "C" {
-#include <ffmpeg/avcodec.h>
+#include <libavcodec/avcodec.h>
 }
 
 #include <math.h>
diff -uNr k3b-1.0.5/plugins/decoder/ffmpeg.orig/k3bffmpegwrapper.cpp k3b-1.0.5/plugins/decoder/ffmpeg/k3bffmpegwrapper.cpp
--- k3b-1.0.5/plugins/decoder/ffmpeg.orig/k3bffmpegwrapper.cpp	2008-05-27 08:23:36.000000000 +0000
+++ k3b-1.0.5/plugins/decoder/ffmpeg/k3bffmpegwrapper.cpp	2008-10-02 12:47:17.000000000 +0000
@@ -18,8 +18,8 @@
 #include "k3bffmpegwrapper.h"
 
 extern "C" {
-#include <ffmpeg/avcodec.h>
-#include <ffmpeg/avformat.h>
+#include <libavcodec/avcodec.h>
+#include <libavformat/avformat.h>
 }
 
 #include <string.h>
@@ -194,7 +194,7 @@
     return i18n("Windows Media v1");
   case CODEC_ID_WMAV2:
     return i18n("Windows Media v2");
-  case CODEC_ID_MP3LAME:
+  case CODEC_ID_MP3:
     return i18n("MPEG 1 Layer III");
   case CODEC_ID_AAC:
     return i18n("Advanced Audio Coding (AAC)");
@@ -290,7 +290,7 @@
 #ifdef FFMPEG_BUILD_PRE_4629
     int len = avcodec_decode_audio( &d->formatContext->streams[0]->codec,
 #else
-    int len = avcodec_decode_audio( d->formatContext->streams[0]->codec,
+    int len = avcodec_decode_audio2( d->formatContext->streams[0]->codec,
 #endif
 				    (short*)d->outputBuffer, &d->outputBufferSize,
 				    d->packetData, d->packetSize );
