--- k3b-0.12.4a/src/option/base_k3bdivxoptiontab.ui.orig	2005-09-10 15:30:45.000000000 +0200
+++ k3b-0.12.4a/src/option/base_k3bdivxoptiontab.ui	2005-09-14 12:53:17.000000000 +0200
@@ -326,7 +326,6 @@
 </widget>
 <layoutdefaults spacing="6" margin="11"/>
 <includehints>
-    <includehint>kdoublenuminput.h</includehint>
     <includehint>knuminput.h</includehint>
     <includehint>knuminput.h</includehint>
     <includehint>knuminput.h</includehint>
