--- k3b-0.11.24/configure.orig	2005-05-11 10:58:16.000000000 +0200
+++ k3b-0.11.24/configure	2005-05-12 00:10:25.000000000 +0200
@@ -33986,7 +33986,7 @@
 fi;
 
 RESMGR_LIB=""
-if test "$ac_cv_use_regmgr" != "no"; then
+if test "$ac_cv_use_resmgr" != "no"; then
 
 
 
--- k3b-0.11.24/configure.in.orig	2005-05-11 10:58:05.000000000 +0200
+++ k3b-0.11.24/configure.in	2005-05-12 00:08:21.000000000 +0200
@@ -284,7 +284,7 @@
 )
 
 RESMGR_LIB=""
-if test "$ac_cv_use_regmgr" != "no"; then
+if test "$ac_cv_use_resmgr" != "no"; then
   KDE_CHECK_HEADERS(resmgr.h, [
          KDE_CHECK_LIB(resmgr,rsm_open_device,[
                  RESMGR_LIB="-lresmgr"
